avm/res/event-grid/system-topic/main.bicep

```diff
@@ -13,8 +13,108 @@ param source string
 @description('Required. TopicType for the system topic.')
 param topicType string
 
+// User-defined types for Event Grid components
+@description('Event subscription destination configuration. Should contain endpointType and properties matching the Azure Event Grid destination schema.')
```

Comment:
The same comment as for the `event-subscription` child-module apply.

Also in general I highly recommend to not create types that just define an objects. That renders the type redundant as it does not provide any added value :)